opm/input/eclipse/Schedule/Well/NameOrder.cpp

```diff
@@ -116,28 +118,50 @@ GroupOrder GroupOrder::serializationTestObject()
 
 void GroupOrder::add(const std::string& gname)
 {
-    auto iter = std::find(this->m_name_list.begin(),
-                          this->m_name_list.end(), gname);
-    if (iter == this->m_name_list.end()) {
-        this->m_name_list.push_back(gname);
+    auto iter = std::find(this->name_list_.begin(),
```

Comment:
Unrelated question because of the linear search: How many entries does name_list usually have?

Reply:
> How many entries does name_list usually have?

For the use case we're discussing here, it's the number of active groups in simulation run.  I've never seen a model in which the number of groups is larger than 50 or so, with fewer than 15-20 being the norm, but I can obviously not rule out corner cases.

> The vector probably needs this particular order, right?

Kind of.  We need this order for restart file output, but I do think we could have a kind of acceleration structure that would enable using binary search here.